### ISSUE TITLE
Add qualifier notation to UmpleOnline

### ISSUE THREAD

**Issue by Nava2** [umple/umple#49]
Originally reported on Google Code with ID 133

---

Add UML qualifier syntax to UmpleOnline once unique on associations is done, which is
issue 88.

---

Reported by @umple on 2011-06-24 20:59:16

**Comment by Nava2** [comment]
Reported by @umple on 2012-06-15 14:53:33
- **Labels added**: Priority-Low, Diffic-Med
- **Labels removed**: Priority-Medium

**Comment by Nava2** [comment]
Reported by @umple on 2012-09-27 20:12:22
- **Owner removed**
- **Blocked on**: #31
- **No longer blocked on**: #31

**Comment by Nava2** [comment]
A qualifier is an ordinary attribute, but it visually appears on the other end of an
association.

For example in the following accountNo is a qualifier because it is an identifier of
the Account in the context of the Bank
class Bank {1 -- \* Account; } 
class Account {accountNo;}
The accountNo would have to be unique with respect to the bank.
The association could also be implemented using a lookup table (hash table) or map,
that mapped account numbers  Accounts for quick access.

A syntax for this could be to combine the attribute and association notations:

class Account {accountNo \* -- 1 Bank;}

This does not resemble UML, but does indicate that the account number is an identifier
at the bank. It could be automatically made unique and generated as a map.

---

Reported by @umple on 2014-03-25 11:48:04

**Comment by TimLethbridge** [comment]
This would need to be done in the grapiviz generated diagrams, which would actually be tricky (and may not be doable). This is low priority, and we may choose to simply not do it.
